### ISSUE TITLE
schema-reader Error: Call to a member function compileTables() on null

### ISSUE THREAD

**Issue by leek** [VemtoOrg/vemto2-issues#27]
**Error Message:**
schema-reader Error: Call to a member function compileTables() on null

**Stack:**
```
schema-reader Error: #0 /Users/leek/Developer/{...}/vendor/laravel/framework/src/Illuminate/Database/Schema/Builder.php(165): Illuminate\Database\Schema\Builder->getTables()
#1 /Users/leek/Developer/{...}/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(355): Illuminate\Database\Schema\Builder->hasTable('media')
#2 /Users/leek/Developer/{...}/database/migrations/2021_08_10_101547_create_media_table.php(11): Illuminate\Support\Facades\Facade::__callStatic('hasTable', Array)
#3 [internal function]: CreateMediaTable->up()
#4 phar:///Applications/Vemto.app/Contents/Resources/static/VMTTL1/classes/MigrationDecoder.php(22): ReflectionMethod->invoke(Object(CreateMediaTable))
#5 phar:///Applications/Vemto.app/Contents/Resources/static/VMTTL1/index.php(77): MigrationDecoder->decode()
#6 phar:///Applications/Vemto.app/Contents/Resources/static/VMTTL1/common/Vemto.php(107): {closure}()
#7 phar:///Applications/Vemto.app/Contents/Resources/static/VMTTL1/index.php(18): Vemto::execute('schema-reader', Object(Closure))
#8 /Applications/Vemto.app/Contents/Resources/static/VMTTL1(15): require('phar:///Applica...')
#9 {main}
```

---

Vemto 2.0.4
darwin 23.3.0
Locale: en-US
App version: 2.0.4

**Comment by TiagoSilvaPereira** [comment]
Hi @leek, could you please provide more details on how we can reproduce this error? Is this a new project or an existing one? Does the 2021_08_10_101547_create_media_table have any problem, as seen in the log? Could you please send the 2021_08_10_101547_create_media_table so we can analyze it?

**Comment by leek** [comment]
@TiagoSilvaPereira This is an existing project and the migration in question is from [Lunar](https://lunarphp.io/):

```
<?php

use Illuminate\Database\Migrations\Migration;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Support\Facades\Schema;

class CreateMediaTable extends Migration
{
    public function up()
    {
        if (! Schema::hasTable('media')) {
            Schema::create('media', function (Blueprint $table) {
                $table->bigIncrements('id');

                $table->morphs('model');
                $table->uuid('uuid')->nullable()->unique();
                $table->string('collection_name');
                $table->string('name');
                $table->string('file_name');
                $table->string('mime_type')->nullable();
                $table->string('disk');
                $table->string('conversions_disk')->nullable();
                $table->unsignedBigInteger('size');
                $table->json('manipulations');
                $table->json('custom_properties');
                $table->json('generated_conversions');
                $table->json('responsive_images');
                $table->unsignedInteger('order_column')->nullable();

                $table->nullableTimestamps();
            });
        }
    }
}
```

Could it be the `Schema::hasTable('media')`?

**Comment by TiagoSilvaPereira** [comment]
@leek I think yes, this is the problem! I'll need to update our schema reader to handle this situation (I'm already projecting a new schema reader to bypass all those different situations)

**Comment by TiagoSilvaPereira** [comment]
Fixed on 2.0.7 by using the new Database Schema Reader mode
